Weather App/UI/SettingsVC/SettingsCoordinator.swift

```diff
@@ -20,9 +22,57 @@ class SettingsCoordinator: Coordinator {
         let viewController = SettingsViewController()
         let viewModel = SettingsViewModel()
         
-        //TODO: setup viewModel
+        viewModel.onEnter = { [weak self] in
+            if let loadedData = self?.loadSettingsData() {
+                viewController.settingsView.setupData(data: loadedData)
+                self?.settingsData = loadedData
+            }
+        }
+        
+        viewModel.onCelsiusTapped = { [weak self] wasChecked in
+            viewController.settingsView.setCelsiusButtonChecked(!wasChecked)
+            viewController.settingsView.setFahrenheitButtonChecked(wasChecked)
+            self?.settingsData.useCelsius = true
+        }
+        
+        viewModel.onFahrenheitTapped = { [weak self] wasChecked in
+            viewController.settingsView.setFahrenheitButtonChecked(!wasChecked)
+            viewController.settingsView.setCelsiusButtonChecked(wasChecked)
+            self?.settingsData.useCelsius = false
+        }
+        
+        viewModel.onHumidityTapped = { [weak self] wasChecked in
+            viewController.settingsView.setHumidityButtonChecked(!wasChecked)
+            self?.settingsData.showHumidity = !wasChecked
+        }
+        
+        viewModel.onPressureTapped = { [weak self] wasChecked in
+            viewController.settingsView.setPressureButtonChecked(!wasChecked)
+            self?.settingsData.showPressure = !wasChecked
+        }
+        
+        viewModel.onWindTapped = { [weak self] wasChecked in
+            viewController.settingsView.setWindButtonChecked(!wasChecked)
+            self?.settingsData.showWind = !wasChecked
+        }
+        
+        viewModel.onExit = { [weak self] in
+            guard let data = try? JSONEncoder().encode(self?.settingsData) else { return }
```

Comment:
Coordinator is accessing UserDefaults?

Reply:
I moved the access to UserDefaults to a separate class.